examples/repository/example_test.go

```diff
@@ -0,0 +1,148 @@
+package repository_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/alpine-hodler/gidari/proto"
+	"github.com/alpine-hodler/gidari/repository"
+	"github.com/alpine-hodler/gidari/tools"
+	_ "github.com/joho/godotenv/autoload"
+)
+
```

Comment:
We should add a function called `TestExamples` to this file that looks like this:

```go
func TestExamples(t *testing.T) {
	t.Cleanup(tools.Quiet()) // This will suppress the `fmt` print statements in CI
	t.Parallel()

	for _, tcase := range []struct{mongoURI string} {
	        {"mongodb://mongo1:27017/coinbasepro"},
         } {
		err := os.Setenv("MONGO_URI", tcase.mongoURI)
		if err != nil {
			t.Fatalf("failed to set environment variable: %v", err)
		}
                
                // Register the example tests
		t.Run("ExampleNew",
			func(t *testing.T) {
				t.Parallel()
				ExampleNew()
			})
			
		t.Run("ExampleTruncate", 
			func(t *testing.T) {
				t.Parallel()
				ExampleTruncate()
			})
			
	        ...
	}
}
```

Since this also uses `os.Setenv` to set the `MONGO_URI`, there is no need to have a .env file; everything will be self-contained with the Docker containers made in `make containers`.